### PR TITLE
Improvements for local development setup process

### PR DESCRIPTION
**Pull request by MateiGrigore**
### Summary:

This PR will improve the setup process for local development of the mongodb-kubernetes-operator. The documentation contained outdated and misleading information.  Also some problems with building images and deploying the operator were encountered because of using a local registry. This

The PR adds:
- Documentation changes to improve the local setup process
- Changes to the `Makefile`:
    - Change the name of the parameter to look for the agent image in the `config.json` file to `agent_image`
    - Add `--ignore-not-found` flag to `uninstall-crd` target because of errors encountered when calling this during the `make deploy` command where no crd was installed yet
    - Add `$(IMG_BUILD_ARGS)` to all image building targets to allow the `--insecure` flag to be passed as a paramter
- Changes to the `inventory.yaml` file:
    - Removed default values for `agent_distro` and `tools_distro`
    - Changed the values for `agent_distro` and `tools_distro` in buildargs to be extracted from the params
- Changes to the `pipeline.py` file:
    - Allow for a new argument `insecure` to be passed
- Chnages to the `setup_kind_cluster.sh`:
    - Add the registry config to the nodes

### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: inventory.yaml
Comment (Julien-Ben) [inline inventory.yaml:12]: Did these default values caused troubles in your setup ?

File: pipeline.py
Comment (Julien-Ben) [inline pipeline.py:90]: I think we should also add a default here.

**Review by Julien-Ben — APPROVED** [review]
Very nice setup improvements ! 👏 
Left a few comments but LGTM overall
(please wait for others to have a look before merging)

File: docs/run-operator-locally.md
Comment (nammn) [inline docs/run-operator-locally.md:25]: can you clarify this part?

File: inventory.yaml
Comment (nammn) [reply inventory.yaml:12]: What did exactly happen? What was the error?

File: scripts/dev/setup_kind_cluster.sh
Comment (nammn) [inline scripts/dev/setup_kind_cluster.sh:76]: can you explain to me this change and why this is necessary? In meko we are not doing this and it works fine

Comment (nammn) [reply scripts/dev/setup_kind_cluster.sh:76]: I personally need more information why we are doing this and what it does before approving. I won't block the PR but if it can wait I prefer to let this one open until mircea is back

File: inventory.yaml
Comment (nammn) [reply inventory.yaml:12]: isn't that because we hardcoded the tools_distr and agent_distro?
But wouldn't it work if you still keep the defaults? 😕 

Nevertheless, works for me!

Comment (nammn) [reply inventory.yaml:12]: yes, we actually had a bug! We pulled that fix out https://github.com/mongodb/mongodb-kubernetes-operator/pull/1596

File: scripts/dev/setup_kind_cluster.sh
Comment (mircea-cosbuc) [reply scripts/dev/setup_kind_cluster.sh:76]: This was basically updating the local registry setup using the official docs guide: https://kind.sigs.k8s.io/docs/user/local-registry/ . This is better because it's not proxying the registry it actually makes the kind node aware of the localhost registry.

**Pull request merged**
